### PR TITLE
chore: use workspace-wide lints

### LINKED ISSUE #3344: Configure lint options in Cargo.toml
### What type of enhancement is this?

Tech debt reduction

### What does the enhancement do?

https://github.com/GreptimeTeam/greptimedb/blob/cf08a3de6b13695a2445fa2edca467d12125bd71/.cargo/config.toml#L8-L15

Now https://github.com/rust-lang/cargo/issues/5034 is resolved. You can see 

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by SteveLauC**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Specify clippy lints in `Cargo.toml` rather than `.cargo/config.toml`, closes #3344 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)

#3344

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -2,14 +2,4 @@
...
-
-[build]
-rustflags = [
-    # lints
-    # TODO: use lint configuration in cargo https://github.com/rust-lang/cargo/issues/5034
-    "-Wclippy::print_stdout",
-    "-Wclippy::print_stderr",
-    "-Wclippy::implicit_clone",
-]
+sqlness = "run --bin sqlness-runner --"
```
Comment (WenyXu) [inline .cargo/config.toml:5]: nit: Add a newline after L5

**Review by WenyXu — APPROVED** [review]

File: .cargo/config.toml (hunk 1 above)
Comment (SteveLauC) [reply .cargo/config.toml]: done

**Comment by SteveLauC** [comment]
Ping me if you guys need me to squash my commits

**Review by tisonkun — APPROVED** [review]
Thank you! We use Merger Queue and Squash Commits. So you don't have to manually squash the commits.

**Pull request merged**